libs/protocol-ui/src/components/molecules/ControlledDatePicker/ControlledDatePicker.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { SyntheticEvent } from 'react';
+import ReactDatePicker, { ReactDatePickerProps } from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { Box } from '@chakra-ui/react';
+import ControlledDatePickerButton from './ControlledDatePickerButton';
+
+export interface ControlledDatePickerProps extends ReactDatePickerProps {
+  onChange: (
+    date: Date[] | Date | [Date | null, Date | null] | null,
+    event: SyntheticEvent<Date, Event> | undefined,
+  ) => void;
+  startDate: Date | null | undefined;
+  endDate: Date | null | undefined;
+  maxDate?: Date | null | undefined;
+}
```

Comment:
Is this necessary? Couldn't we use ReactDatePickerProps? Some of these look like what is already in there.

Reply:
Ah, yeah. That's a good point. Let me try that -- I had some issues with this originally but I think it may have been from something else. If these are unnecessary I'll update our other DatePicker as well 👍

Reply:
Works since spreading props and extending -- changed the DatePicker to use this pattern